### PR TITLE
fix: fixes online->offline race condition

### PR DESCRIPTION
**Pull request by stalniy**
## Why

Closes CON-437

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced provider offline marking mechanism to use timestamp-based validation, ensuring providers are accurately marked offline only when their last activity falls before a specified cutoff time.

* **Tests**
  * Extended integration and unit test coverage for timestamp-based provider offline operations, including edge cases and accuracy verification.

### CODE REVIEW

File: apps/provider-inventory/src/services/stream-lifecycle-manager/stream-lifecycle-manager.service.ts [hunk 1]
```diff
@@ -71,10 +71,11 @@ export class StreamLifecycleManagerService {
...
     this.#offlineDataloader = new Dataloader(
-      async owners => {
-        await this.#inventoryRepo.bulkMarkOffline(owners as string[]);
-        this.#logger.info({ event: "PROVIDERS_MARKED_OFFLINE", owners });
-        return Array.from(owners, () => null);
+      async keys => {
+        const owners = keys.map(k => k.owner);
+        const results = await this.#inventoryRepo.bulkMarkOffline(owners, keys[0].requestedAt);
+        const updatedOwners = new Set(results.map(r => r.owner));
+        return Array.from(owners, owner => updatedOwners.has(owner));
```
[the remark this answers is not in the document]
Comment (stalniy) [reply apps/provider-inventory/src/services/stream-lifecycle-manager/stream-lifecycle-manager.service.ts:78]: it's just 1 second, it's accepted risk

**Review by ygrishajev — APPROVED** [review]
